### PR TITLE
Feature/landing page search

### PR DESCRIPTION
**Pull request by MMostella**
## Description

In this PR, we added a Landing Search Page that displays if the user does not have any current bookmarks. In which case, The User will be pushed to a standard search bar landing page. Upon creating/adding bookmarks, The user will have their bookmarks display on initial app launch.

We added Titles for the Bookmark page and Search Directory, depending on the results you are being shown. Also, if/when a user searches for a word or phrase that has no returns results, we added a alert that notifies the user of no result and leaves them on the page they were currently on at the time of the search.

https://www.loom.com/share/03b3dbf379f444d59a50ed59313b83e3

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by mgreenwaldm — APPROVED** [review]

File: src/state/actions/index.js [hunk 1]
```diff
@@ -32,6 +39,7 @@ export const getDocs = authState => async dispatch => {
       dispatch({ type: SET_DOCS, payload: Response });
     } else {
       dispatch({ type: FINISH_FETCH });
+      dispatch({ type: SEARCH_BAR });
     }
   } catch (err) {
     console.log(err);
```
Comment (christina-yun) [inline src/state/actions/index.js:45]: Should this be just a console.error or were you planning on making this into something that plugs into a display of the error for the user?

File: src/state/actions/index.js [hunk 2]
```diff
@@ -42,7 +50,13 @@ export const searchDocs = (search, authState) => dispatch => {
...
-      dispatch({ type: SET_DOCS, payload: data.Response });
+      if (data.Response.length === 0) {
+        alert('No search results');
+        dispatch({ type: FINISH_FETCH });
+      } else {
+        dispatch({ type: SET_DOCS, payload: data.Response });
+        dispatch({ type: SEARCH, payload: search });
+      }
     })
     .catch(console.error);
```
Comment (christina-yun) [inline src/state/actions/index.js:61]: Same questions here re: console.error

**Review by christina-yun — APPROVED** [review]
Amy, big fan of you walking us through the functionality of the code you wrote. Would also have loved to see in addition to how the app is working, what those choices were in the code. Overall though, this is super impressive.

I commented a bit on the console.error but I don't think that should necessarily prevent merge, but something to think about so that errors could be known to the user if things go wrong.

**Review by Klove-A — APPROVED** [review]
The quick work on this guys, it looks good. Implements a needed feature, clear description , a video walk-through of the code in action and numerous commits with meaningful messages. My two notes are to try and remember to write commits in present tense (adds not added) and it would've been nice to see the no search results alert demonstrated in the video.

File: src/state/actions/index.js (hunk 1 above)
Comment (Klove-A) [reply src/state/actions/index.js:45]: Please someone who actually worked on this correct me if I'm wrong, but I believe that's what they're using to get the no search results pop up.

File: src/components/common/Header.js [hunk 3]
```diff
@@ -64,17 +64,23 @@ function MainHeader(props) {
...
-          className="search_bar"
-          placeholder="Search"
-          onSearch={onSearch}
-        />
-        <Button onClick={listView}>List</Button>
-        <Button onClick={thumbnailView}>Thumbnail</Button>
-        <Link to="/">
-          <Button type="default">Bookmarks</Button>
-        </Link>
+        {props.page === 'bar' ? (
```
Comment (stu-wd) [inline src/components/common/Header.js:67]: we should destructure this given that we are doing it with `searchDocs` early on

Comment (stu-wd) [inline src/components/common/Header.js:67]: we need to destructure this as we did with `searchDocs`

**Comment by stu-wd** [comment]
As mentioned this morning by Paul, I think we should move the `Logout` button to the right side of the new starting page for continuity.

Is there any way that we can connect the "Directory" to state, so that we get something like "Results for 'whatever-user-searched'"?

There's a mixture of both `rem` and `px` in some of the styling. As mentioned in another PR, this can be something that we tackle next week to create styling continuity across the entire app.

I appreciate the frequent commits, as it helps us to see where your brain was along the way. Like I said this morning when I saw it, this thing looks FRESH! It's a shame that they will only see it if they don't have bookmarks because it is plain out visually enjoyable.

File: src/state/actions/index.js (hunk 1 above)
Comment (MMostella) [reply src/state/actions/index.js:45]: Myself or a UX will be adding a Ant Design model in replace of the alert for better looking user experience tomorrow 12-15

**Pull request merged**